src/transformers/modeling_gpt2.py

```diff
@@ -946,3 +946,111 @@ def forward(
             hidden_states=transformer_outputs.hidden_states,
             attentions=transformer_outputs.attentions,
         )
+
+
+@add_start_docstrings(
+    """The GPT2 Model transformer with a sequence classification head on top
+    (linear layer). """,
+    GPT2_START_DOCSTRING,
+)
+class GPT2ForSequenceClassification(GPT2PreTrainedModel):
+    authorized_missing_keys = [r"h\.\d+\.attn\.masked_bias", r"lm_head\.weight"]
+
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+        self.transformer = GPT2Model(config)
+        self.score = nn.Linear(config.n_embd, self.num_labels, bias=False)
```

Comment:
Other models use two layers for the classification head. We sure we want only one?

Reply:
Adding another layer for the classification head would make this head incompatible with `DialogRPT`, which is the reason this PR is done.

There is no other interesting pre-trained sequence classification GPT-2 model that I know of, but if we hear of one that uses a second layer later on, we can find a way to enable that.